7.18.2 regression: react/jsx-indent lots of false positives

Since the last version 7.18.2, eslinting [cockpit](https://github.com/cockpit-project/cockpit/) gets tons of false positives like these:
```
/var/home/martin/upstream/cockpit/pkg/systemd/init.js
  1326:20  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1327:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1327:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1328:21  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1329:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
```

on this code, which seems perfectly well indented:
```
    function create_timer_file() {
        var unit = "[Unit]\nDescription=";
        var service = "\n[Service]\nExecStart=";
        var timer = "\n[Timer]\n";
        var install = "[Install]\nWantedBy=timers.target\n";
        var service_file = unit + timer_unit.Description + service + timer_unit.Command + "\n";
```
The dozens of other errors are also wrong. Things were fine until 7.18.0, I haven't tested 7.18.1 yet.

--- a/lib/rules/jsx-indent.js
+++ b/lib/rules/jsx-indent.js
@@ -374,6 +374,9 @@ module.exports = {
       if (!node.parent) {
         return;
       }
+      if (node.parent.type !== 'JSXElement' && node.parent.type !== 'JSXFragment') {
+        return;
+      }
       const parentNodeIndent = getNodeIndent(node.parent);
       checkLiteralNodeIndent(node, parentNodeIndent + indentSize);
     }

--- a/tests/lib/rules/jsx-indent.js
+++ b/tests/lib/rules/jsx-indent.js
@@ -989,6 +989,14 @@ const Component = () => (
       '</App>'
     ].join('\n'),
     options: ['tab']
+  }, {
+    // don't check literals not within JSX. See #2563
+    code: [
+      'function foo() {',
+      'const a = `aa`;',
+      'const b = `b\nb`;',
+      '}'
+    ].join('\n')
   }],
 
   invalid: [{
@@ -2002,6 +2010,21 @@ const Component = () => (
     ].join('\n'),
     errors: [
       {message: 'Expected indentation of 1 tab character but found 2.'}
+    ]
+  }, {
+    code: [
+      '<>',
+      'aaa',
+      '</>'
+    ].join('\n'),
+    parser: parsers.BABEL_ESLINT,
+    output: [
+      '<>',
+      '    aaa',
+      '</>'
+    ].join('\n'),
+    errors: [
+      {message: 'Expected indentation of 4 space characters but found 0.'}
     ]
   }]
 });

